### PR TITLE
 add history append subcommand 

### PR DESCRIPTION
**Pull request by Accurate0**
## Description

This adds the history subcommand for `append` to allow history items to be added without executing the command.

Fixes issue #4506

## TODOs:

- [x] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [x] User-visible changes noted in CHANGELOG.rst

### CODE REVIEW

File: src/builtins/history.rs [hunk 1]
```diff
@@ -343,6 +346,13 @@ pub fn history(parser: &Parser, streams: &mut IoStreams, args: &mut [&wstr]) ->
             history.save();
         }
         HistCmd::HIST_UNDEF => panic!("Unexpected HIST_UNDEF seen"),
+        HistCmd::HIST_PUSH =>
+        {
+            #[allow(clippy::unnecessary_to_owned)]
+            for arg in args.iter().copied() {
+                history.add_commandline(arg.to_owned());
+            }
```
Comment (Xiretza) [inline src/builtins/history.rs:354]: No need to silence the lint:

```suggestion
            for &arg in args {
                history.add_commandline(arg.to_owned());
            }
```

**Review by Xiretza — APPROVED** [review]
Looks good to me overall except for a small nit; could you squash the commits down to one (or maybe an extra one for the whitespace changes) as well?

File: src/builtins/history.rs (hunk 1 above)
Comment (Accurate0) [reply src/builtins/history.rs:354]: I get this without silencing the lint, same as the `HIST_DELETE`

Comment (Xiretza) [reply src/builtins/history.rs:354]: Yes, if you follow the clippy suggestion and then fix the type error (oops, clippy bug), you get `for &arg in args` as in my suggestion.

Comment (Accurate0) [reply src/builtins/history.rs:354]: Oh sorry, I didn't notice the change, thanks. Will fix.

**Comment by krobelus** [comment]
LGTM. Should add the new command to `share/completions/history.fish`.
I wonder if we ever want to insert in the middle, I guess that's unlikely

**Comment by krobelus** [comment]
I guess we could call it `append` so it's clear that it's not a stack of history files.. though I'm not sure if that matters

**Comment by Accurate0** [comment]
I've added it to the completions as push, let me know if you'd prefer it changed to append. I went off the original suggestion in the issue

**Comment by krobelus** [comment]
I think I prefer "append" because that's the term used for opening files with `>>`, and history is still a text file.
It also fits `set --append` which we use for variables. Note that we  also have `$history` which is in reverse order so to append we'd want to use `--prepend`.. but the variable is read-only, fortunately.
I don't expect that we want it writable because misuse would be destructive.

**Comment by Accurate0** [comment]
> I think I prefer "append" because that's the term used for opening files with `>>`, and history is still a text file. It also fits `set --append` which we use for variables. Note that we also have `$history` which is in reverse order so to append we'd want to use `--prepend`.. but the variable is read-only, fortunately. I don't expect that we want it writable because misuse would be destructive.

Seems fair to me, I've made the changes and squashed to keep the commits clean

**Comment by Accurate0** [comment]
Check failure doesn't look related to my changes, let me know if you disagree otherwise it might just need a rerun

**Pull request merged**